src/pickWithDefault.js

```diff
@@ -0,0 +1,36 @@
+var _curry3 = require('./internal/_curry3');
+
+
+/**
+ * Similar to `pick` except that this one includes a `key: <default>` pair for
+ * properties that don't exist
+ *
+ * **Note:** A property set to undefined will be left untouched.
+ *
+ * @func
+ * @memberOf R
+ * @since v0.22.2
+ * @category Object
+ * @sig [k] -> * -> {k: v} -> {k: v}
```

Comment:
`StrMap` is not really a Sanctuary construct. In fact, I got the idea from [PureScript][1].

JavaScript objects are used in two distinct ways: as records (with known fields of known—possibility heterogeneous—types), and as dictionaries (with string keys and homogeneous values). In the latter case we need a unary type so we can indicate the type of the dictionary's values. This is what `StrMap` provides.

I agree that if we're to mention `StrMap` in type signatures we should document it in Ramda itself.


[1]: https://pursuit.purescript.org/packages/purescript-maps/1.1.0/docs/Data.StrMap